server/src/com/cloud/template/TemplateAdapterBase.java

```diff
@@ -339,9 +341,13 @@ protected VMTemplateVO persistTemplate(TemplateProfile profile, VirtualMachineTe
             new VMTemplateVO(profile.getTemplateId(), profile.getName(), profile.getFormat(), profile.getIsPublic(), profile.getFeatured(), profile.getIsExtractable(),
                 profile.getTemplateType(), profile.getUrl(), profile.getRequiresHVM(), profile.getBits(), profile.getAccountId(), profile.getCheckSum(),
                 profile.getDisplayText(), profile.getPasswordEnabled(), profile.getGuestOsId(), profile.getBootable(), profile.getHypervisorType(),
-                profile.getTemplateTag(), profile.getDetails(), profile.getSshKeyEnabled(), profile.IsDynamicallyScalable());
+                profile.getTemplateTag(), profile.getDetails(), profile.getSshKeyEnabled(), profile.IsDynamicallyScalable(), profile.isDirectDownload());
```

Comment:
`profile.isDirectDownload()` might return null since all constructors don't set the direct download `Boolean` variable. The method declaration expects a `boolean`, so no `null`.

Reply:
Done, thanks